### PR TITLE
Adds soapy water as a weak disinfectant

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
```SUMMARY: Content "Adds soapy water as a weak disinfectant"```

#### Purpose of change
Issue #25628 inspired me to add a weak disinfectant in form of **soapy water**, that you can use to clean your wounds.

#### Describe the solution
Added item and recipe for it.

#### Describe alternatives you've considered
I've considered giving it strength of 1, but I think clean water can be strength 1, then soapy water 2, then makeshift (ethanol) disinfectant 3, then medical-grade disinfectants at 4. I have not added any disinfecting qualities to clean water (yet) before a proper discussion on that takes place.

#### Additional context
N/A

### CODE REVIEW

**Comment by chaisawlajatang** [comment]
IMHO we already have an abundance of disinfectant items - a bottle with 10 (!) charges in first aid kits (also in ambulance or on dead bodies), makeshift disinfectant with 10 charges (requires ZERO skills), etc. It makes infected wounds an unsignificant threat.

**Comment by Xhuis** [comment]
Would detergent also work in place of soap or would that worsen the wound?

**Comment by nexusmrsep** [comment]
@chaisawlajatang while I agree with you that amount of charges in high-grade disinfectants is questionable (although many would say disintectants are hard to find), this has nothing to to with this since one unit of clean water + 1 soap charge = 1 unit of soapy water that has no extra charges. With 1 soap (10 charges) and 5 bottles of clean water you will get only 10 uses of weak soap water.

@Xhuis this is what I was asking myself too. On one hand survivors are not choosers but on the other hand: [source 1](https://www.ncbi.nlm.nih.gov/pubmed/2301881) or [source 2](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4141917/) suggests it's too aggressive. Still not entirely convinced to add it.

**Comment by Qrox** [comment]
In the second source the chemical burns are caused by undiluted detergent and prolonged exposure without proper care, so the survivor may reduce the adverse effects by carefully applying the detergent. And if we model the "adverse tissue reaction" or "chemical burns" as mentioned in the two sources (as acute pain perhaps?), maybe detergents can serve in a pinch if the survivor has no other choices. But given the abundance of disinfectants currently in the game they may only become useful in the early game and in very few playthroughs.

**Comment by Regularitee** [comment]
Some quick facts about soaps:

Soaps are generally molecules that can bind to oil on one end, water on the other. This is why soaps allow you to"wash away" oil from dishes using tap water, despite the fact oil normally repels water.

**Traditional soaps** clean the hands by virtue of its lubrication; it's value in hygiene in is purely that of causing filth and microbes "fall off" rather than actually kill them. So I suspect would be largely useless on wounds, as they could cause said bacteria simply to better slide into the wound.

**Antimicrobial soaps** differ in that their purpose goes beyond just attracting oil and water to one another. Instead, they also contain chemical compounds (usually triclosan) to actively destroy bacteria. These are a obviously harder to make yourself, or find, since they're usually have advanced chemical compounds. They will also damage the human body -- except the skin, which has several layers of keratin-filled dead cells acting as a barrier. The rest of the human body (such as open wounds) do not. They are still effective at reducing infection, however.

**Detergents** are soaps not designed regular exposure to human tissue, so are concentrated far beyond levels considered safe for consistent human contact. Modern detergents have become extremely deadly, as the competition to produce more potent detergents have resulted in highly concentrated solutions. These concentrated detergents will almost instantly necrotize exposed (non-skin) cells on contact. Their ability to attract oil and water is so powerful they will rip open cell membranes and cause them to spill out their innards (hence, nearly-instant necrosis on contact). Even if you were to dilute them, I suspect they would be not much more useful than traditional soaps due to lacking antimicrobial compounds.

And in case anyone is curious on the topic of wound sterilization: a bonus.

**Cauterization** is an alternative to chemical sterilization has fallen out of medical favour in the last few decades. While the heat will kill bacteria on the wound, it also leaves the body cells in and around the wound greatly weakened and damaged, making future infections much more likely.

**Comment by nexusmrsep** [comment]
> But given the abundance of disinfectants currently in the game they may only become useful in the early game and in very few playthroughs.

**Comment by Regularitee** [comment]
If you want the best combination of realism and gameplay, I'd simply create a separate type of soaps: Antimicrobial soaps (as hand sanitizer and/or dish soap), a rarer version of normal soap. I think regular soap would be all but useless on existing wounds. The only realistic way they could help prevent infection would be the sterilization of tools and gloves before a surgery. 

That said, the game stretches realism in many other places, so I would not oppose you if you did go ahead and give conventional soaps some disinfectant value. Maybe under the assumption that "soap" is an abstraction of all soaps (both regular and antimicrobial), or something like that..

**Comment by Marrim** [comment]
Percentage chances make little difference unless there is some kind of cap in place. And how can you tell whether you "succeeded" at cleaning the wound? You can only know when it either starts healing, or begins to putrefy.

Every unsuccessful attempt should diminish your further chances to clean the wound successfully. Then those low percentages on low-quality disinfectant will start to matter.

**Comment by FulcrumA** [comment]
I like the idea of introduction of higher-quality, semi-common in grocery shop, some homes and medical places antimicrobial soap - which would offer absolutely same options regular soap does, on top of being usable in recipe of mixing it with water for characters with the very basic (1) of First Aid.

**Comment by Regularitee** [comment]
Realism aside, I do have some concerns about how it might affect game balance if a disinfectant crafting were to be available practically from the game start. I fear we may end up with something like makeshift bandages are now: materials are so common as to make infinite numbers available even from the very start, which in turn may prompt a nerfing of that type of item itself to prevent widespread abuse or breaking game balance.

I do know soaps aren't necessarily as common as rags currently are, but I can't say with certainty they are rare enough as to avoid upsetting game balance, by making disinfectant much more accessible early-game. Especially when each bar of soap (10 charges) nearly guarantees curing an infection. I'd rather make the entire bar of soap produce a single dose of low-success disinfectant, if this must be done.

**Comment by FulcrumA** [comment]
I share @Regularitee concern, but I'd still maintain that making a different, antimicrobial soap that would have regular soap's functionality (perhaps with lesser amounts required to wash stuff) and could be turned into soapy antimicrobial water - would take care of the balancing issue while still allowing the item to be implemented.

**Comment by nexusmrsep** [comment]
> Realism aside, I do have some concerns about how it might affect game balance if a disinfectant crafting were to be available practically from the game start. I fear we may end up with something like makeshift bandages are now: materials are so common as to make infinite numbers available even from the very start, which in turn may prompt a nerfing of that type of item itself to prevent widespread abuse or breaking game balance.

1. Soap is not that common to find (as you said). That's a first limiting factor. Other disinfectants are in fact too common, as mentioned above.
2. You can make only 10 charges (or uses) from 1 soap. That's a second limiting factor.
3. There is "bite heal" factor that can be set from 0-100%  Here set on 25% Third limiting factor.
4. There is "potency" factor (explained above). Here set on 2. Fourth limiting factor.

Just tweaking above values should help (perhaps not touching nr 1 as soap is used for washing too).

I'm repeating myself here: if you don't like the values I've set for soapy water:
- propose other values,
- run and provide statistics on it's potency,
- explain proposed values and convince me.

For now I still think it's not affecting balance. Nor is makeshift bandages. The new concept is to use them as often as you can, so their availability is not a concern. Its a feature.

I don't think adding a separate soap is a good idea. It might be more effective (lvl 3?) but this does not negate validity of soapy water. Why do you wash your hands with soap, especially before eating? Not to clean it from dirt and bacteria so they don't get to your digestive system? Then why wouldn't you use it to clean dirty wound or at least it's vicinity to protect it the same way? Furthermore most first aid manuals, like for ex. [HERE](http://www.ilkyardim.org.tr/dokumanlar/EFAM_printonly.pdf) advice on rinsing wounds with clean water.

**Comment by kevingranade** [comment]
I'm much more in favor of using clean water than soapy water as a wound cure.

I'm against adding any low probability treatments as long as the system works the way it does now where each treatment has an independent chance of succeeding.  It creates a bad situation where large numbers of applications can always stand in for high quality cures.

I think the balance issue of introducing common but unreliable cures could be avoided by making the curing effect deterministic instead of random.

In other words, there isnt an x% chance to cure per use, instead the *wound* has a x% chance of being curable with a given treatment, and that doesn't change for the lifetime of the wound.

The way antibiotics work has a similar effect where you can't just throw more doses at it, but that "chance to heal over time" system doesnt make sense for wound cleansing.

**Comment by nexusmrsep** [comment]
@kevingranade 
> In other words, there isn't an x% chance to cure per use, instead the wound has a x% chance of being curable with a given treatment, and that doesn't change for the lifetime of the wound.

To be clear: do you mean a mechanic where a wound will have this % value for each kind of disinfectant, but after first use of that disinfectant the effect would be binary and constant, so the wound was either disinfected or not, and if not the same disinfectant will not work again, regardless of number of uses?

> The way antibiotics work has a similar effect where you can't just throw more doses at it, but that "chance to heal over time" system doesn't make sense for wound cleansing. 

"Chance to heal overtime" or what I called "potency factor" above AFAIK has nothing to do with removing "bite" effect (so its not like antibiotics at all) but only adds healing speed bonus like bandages. That to my belief is simulating the fact that clean wound can heal faster, as there is no dirt and bacteria to hamper the healing process. From overall concept of wound healing in present state of the game it seems desirable to always clean and bandage wounds to promote healing. @Vasyan2006  can you confirm if I'm right here?
image
Do you want that effect removed in target healing system?

I'll be closing this PR after above will be answered for the record.

**Comment by kevingranade** [comment]
> To be clear: do you mean a mechanic where a wound will have this % value for each kind of disinfectant, but after first use of that disinfectant the effect would be binary and constant, so the wound was either disinfected or not, and if not the same disinfectant will not work again, regardless of number of uses?

Exactly that, it might be a pre-determined severity of the wound, and different treatments are capable of curing different severity levels, or it might still be random, but an effect is applied that says a certain treatment has already been tried.

The goal is that each treatment has one shot at working, and if it fails, number of applications doesnt matter.

**Comment by FulcrumA** [comment]
Maybe antibiotic effect could be added as a quality, like in case of tools, with potentially infected wounds having semi-randomized severity requiring various levels of antibiotics. Could be also useful as a json then, allowing streamlined creation of monsters with (statistically) more or less infectious bite.

**Comment by Vasyan2006** [comment]
> From overall concept of wound healing in present state of the game it seems desirable to always clean and bandage wounds to promote healing.

Yes, having both healing effects from bandages and disinfectant will improve HP recovery, because open wounds will recover faster if no bacterias are around.
But it is separated from initial cleaning the wound. Success chances of cleaning depend only on `"bite" : 0.95` properties of healing items.

>Do you want that effect removed in target healing system?

Their is no reason to disinfect a bruise, so different wounds will requiere different treatment.
